conans/test/integration/toolchains/cmake/test_cmake.py

```diff
@@ -31,3 +31,52 @@ def run(self, *args, **kwargs):
     assert "-something" in client.out
     assert "--testverbose" in client.out
     assert "-testok" in client.out
+
+def test_build_type_single_config_warn():
+    client = TestClient()
+
+    conanfile = textwrap.dedent("""
+        from conan import ConanFile
+        from conan.tools.cmake import CMake
+        class Pkg(ConanFile):
+            name = "pkg"
+            version = "0.1"
+            settings = "os", "compiler", "arch"
+            generators = "CMakeToolchain"
+            def build(self):
+                cmake = CMake(self)
+                cmake.configure()
```

Comment:
This build, in single-config build systems is not using ``build_type`` at all. What is the purpose of doing a CMake build without ``build_type``, which is explicitly discouraged by ``CMake``?

Reply:
How can I specify a build type then in this case a profile does not specify any build_type? 
Using `cmake.build(build_type="Release")` does not have any effect (see test above). 
Shall I use `tc.cache_variables`?

Reply:
Why not defining the ``build_type`` in the profile?

Reply:
Refering to settings.yml the `build_type` MAY be empty. So we must cover this case as well.